wasmcloud-wasi-kv/wasmcloud-wasi-kv-wit/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+    /// Get a value from the key value store
+    fn get(_bucket: Bucket, key: String) -> Result<IncomingValue, Error> {
+        println!("[debug][kv-provider] performing get...");
+
+        let mut encoder = wasmbus_rpc::cbor::vec_encoder(false);
+        encoder.str(&key)
+            .map_err(|_| ERROR_INTERNAL_PROVIDER_REQUEST_SERIALIZATION_FAILURE)?;
+        let payload = encoder.into_inner();
+
+        match wasmcloud_host_call(
+            WASMCLOUD_DEFAULT_BINDING,
+            WASMCLOUD_KV_NAMESPACE,
+            "KeyValue.Get",
+            Some(&payload),
+        ) {
+            Ok(Some(resp_bytes)) => {
+                // Parse the get response
+                let resp = wasmbus_rpc::common::deserialize(&resp_bytes)
+                    .map_err(|_| ERROR_INTERNAL_PROVIDER_RESPONSE_DESERIALIZATION_FAILURE)?;
+                match resp {
+                    // TODO(QUESTION) what do they expect?
+                    GetResponse { exists: false, .. } => Err(ERROR_NO_SUCH_KEY),
+                    GetResponse {
+                        exists: true,
+                        value,
+                    } => {
+                        let iv = create_incoming_value(value.into())?;
+                        Ok(iv)
+                    },
+                }
+            }
+            _ => Err(ERROR_INTERNAL_UPSTREAM_OPERATION_FAILURE),
+        }
+    }
+
+    /// Set a value from the key value store
+    fn set(_bucket: Bucket, key: String, outgoing_value: OutgoingValue) -> Result<(), Error> {
+        println!("[debug][kv-provider] performing set...");
+
+        // Read the value from the outgoing value handle
+        //
+        // TODO(QUESTION): It's not strictly necessary to drop an OV after it's used...
+        // It is also a valid use-case to submit the same OV for multiple keys...
+        // Can we trust consumers to requst cleanup of values?
+        let ov_bytes = clone_hashmap_value(&OUTGOING_VALUES, outgoing_value)?.unwrap_or(Vec::new());
+
+        // Build payload
+        let mut encoder = wasmbus_rpc::cbor::vec_encoder(false);
+        encode_set_request(
+            &mut encoder,
+            &SetRequest {
+                key,
+                // TODO(ISSUE): wasmcloud KV contract expects a String, not just bytes
+                // Rust does not just send bytes as a String -- and trying to is unsafe
+                // https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_unchecked
+
+                // value: unsafe {
+                //     String::from(
+                //         from_utf8_unchecked(&ov_bytes)
+                //             .map_err(|_| ERROR_INTERNAL_PROVIDER_REQUEST_SERIALIZATION_FAILURE)?
+                //     )
+                // },
+                value: String::from(
+                    from_utf8(&ov_bytes)
+                        .map_err(|_| ERROR_INTERNAL_PROVIDER_REQUEST_SERIALIZATION_FAILURE)?,
+                ),
```

Comment:
I'd extract it into a separate `value` binding and use `.into()` rather than the outer `from` call

Reply:
Oh does that solve the issue with the `String` conversions that Rust does, or is that more of a stylistic change?